integrated_channels/xapi/serializers.py

```diff
@@ -83,7 +92,23 @@ def get_course_duration(self, obj):
         Returns:
             (timedelta): Duration of a course.
         """
-        duration = obj.end - obj.start if obj.start and obj.end else None
-        if duration:
-            return strfdelta(duration, '{W} weeks {D} days.')
-        return ''
+        course_run = None
+        duration = None
+        site = self.context.get('site')
+        if site:
+            cache_key = get_cache_key(course_id=obj.id, site=site)
+            cached_response = TieredCache.get_cached_response(cache_key)
+            if cached_response.is_found:
+                course_run = cached_response.value
+            else:
+                try:
+                    _, course_run = CourseCatalogApiServiceClient(site).get_course_and_course_run(obj.id)
+                    TieredCache.set_all_tiers(cache_key, course_run, CACHE_TIMEOUT)
+                except ImproperlyConfigured:
+                    LOGGER.warning('CourseCatalogApiServiceClient is improperly configured.')
+        if course_run and course_run.get('max_effort') and course_run.get('weeks_to_complete'):
+            duration = '{effort} hours per week for {weeks} weeks.'.format(
```

Comment:
I think `{min_effort} to {max_effort} hours per week for {weeks} weeks.` would be better, what do you say?